public/app.js

```diff
@@ -37,14 +53,22 @@ class Controller {
             DataController.detachSnapshotListener();
             console.log(e.target.innerText);
             let project = e.target.innerText;
-            // Add the graphs container to the UI
-            UIController.addGraphs(project);
-            // Update and show the Graphs
-            let unsubscribeFunc = DataController.watchProjectTrafficData(
-                project,
-                GraphController.updateGraphs
-            );
-            DataController.registerSnapshotListener(unsubscribeFunc);
+            location.hash = project;
+            Controller.displaySelectedProject(project);
+        }
+    }
+
+    static deepLinktoPage(activeProjectsData) {
+        let activeProjects = [];
+        activeProjectsData.forEach(project => {
+            activeProjects.push(project.project_name);
+        });
+        if (activeProjects.includes(Controller.hash)) {
+            let project = Controller.hash;
+            Controller.displaySelectedProject(project);
+        } else {
+            history.replaceState(null, null, " ");
```

Comment:
Could you add a brief comment justifying this?

Reply:
Added